samtranslator/model/eventsources/push.py

```diff
@@ -921,7 +924,29 @@ def _add_swagger_integration(  # type: ignore[no-untyped-def] # noqa: too-many-s
                 path=self.Path, method_name=self.Method, request_parameters=parameters
             )
 
-        api["DefinitionBody"] = editor.swagger
+        if api.get("MergeDefinitions", False):
+            self._merge_definitions(api, editor.swagger)
+        else:
+            api["DefinitionBody"] = editor.swagger
+
+    def _merge_definitions(self, api: Dict[str, Any], generated_body: Dict[str, Any]) -> None:
+        """
+        Merge SAM generated swagger definition into inline DefinitionBody using the follow rule:
+        - for a conflicting key, use SAM generated value
+        - otherwise include key-value pairs from both definitions
+        """
+        paths_property = api["DefinitionBody"]["paths"]
+
+        if self.Path not in paths_property:
```

Comment:
do we have validation on the type of `paths_property` (to be a dict) anywhere since it is a customer input?